packages/syft/src/syft/client/api.py

```diff
@@ -582,6 +582,20 @@ def unwrap_and_migrate_annotation(annotation: Any, object_versions: dict) -> Any
         return migrated_annotation[0]
 
 
+def result_needs_api_update(api_call_result: Any) -> bool:
+    # relative
+    from ..service.request.request import Request
+    from ..service.request.request import UserCodeStatusChange
+
+    if isinstance(api_call_result, Request) and any(
+        isinstance(x, UserCodeStatusChange) for x in api_call_result.changes
+    ):
+        return True
+    if isinstance(api_call_result, SyftSuccess):
```

Comment:
This will result in refreshing API more frequently since a couple of APIs return SyftSuccess but don't require a refresh.
Should we have a class var at SyftSuccess to indicate if refresh is required ?